src/ucp/rndv/proto_rndv.c

```diff
@@ -617,6 +619,18 @@ ucp_proto_rndv_ack_init(const ucp_proto_common_init_params_t *init_params,
                                     UCS_LINEAR_FUNC_ZERO, perf_p);
 }
 
+static void
+ucp_proto_rndv_report_fast_cmpl(const ucp_proto_multi_init_params_t *params)
+{
+    uint32_t op_attr_mask = ucp_proto_select_op_attr_unpack(
+            params->super.super.select_param->op_attr);
+
+    if ((op_attr_mask & UCP_OP_ATTR_FLAG_FAST_CMPL)) {
+        ucs_warn("RNDV is initialized with FAST_CMPL which is unexpected. "
```

Comment:
maybe `ucs_debug` since fast_cmpl is best effort and UCP users should (generally) not even be aware that a multi step protocol is used?

Reply:
That warning means that user passed FAST_CMPL flag to recv operation and in case of choosing RNDV it can cause undesired performance prediction change. So I would prefer to leave it as warning so users can see that and remove FAST_CMPL from recv operation.